romancal/resample/l3_wcs.py

```diff
@@ -58,34 +60,22 @@ def assign_l3_wcs(model, wcs):
         l3_wcsinfo.dec_corn4 = footprint[3][1]
         l3_wcsinfo.s_region = compute_s_region_keyword(footprint)
 
-    # Fill out wcs-general information
-    try:
-        l3_wcsinfo.x_ref = -transform["crpix1"].offset.value
-        l3_wcsinfo.y_ref = -transform["crpix2"].offset.value
-    except IndexError:
-        log.warning(
-            "WCS has no clear reference pixel defined by crpix1/crpix2. Assuming reference pixel is center."
-        )
-        l3_wcsinfo.x_ref = pixel_center[0]
-        l3_wcsinfo.y_ref = pixel_center[1]
+    log.info(f"transform {transform}")
+    l3_wcsinfo.x_ref = transform.crpix[0]
+    l3_wcsinfo.y_ref = transform.crpix[1]
 
     world_ref = wcs(l3_wcsinfo.x_ref, l3_wcsinfo.y_ref, with_bounding_box=False)
     l3_wcsinfo.ra_ref = world_ref[0]
     l3_wcsinfo.dec_ref = world_ref[1]
 
-    try:
-        cdelt1 = transform["cdelt1"].factor.value
-        cdelt2 = transform["cdelt2"].factor.value
-        l3_wcsinfo.pixel_scale = (cdelt1 + cdelt2) / 2.0
-    except IndexError:
-        l3_wcsinfo.pixel_scale = compute_scale(wcs, world_ref)
```

Comment:
This is another exception along the lines of x_ref / y_ref above that I think we'd like to preserve.